### ISSUE TITLE
Unable to add server in TV client

### ISSUE THREAD

**Issue by supernovae01** [shadowsocks/shadowsocks-android#2052]
I have a Nvidia Shield TV running Oreo. I upgraded from v4.5.7 to the latest v4.6.4. 
Immediately after upgrading, I noticed there were no way to add server information manually. And the import function does not seem to work either. These combined, practically rendered this release unusable. I have reverted back to v4.5.7.
The UI design in the new TV dedicated version is so much more user friendly when using a remote control. It's a shame to see it not working as it should. I hope this minor issue could be fixed in the next release.

Please read contributing guidelines. Thanks.

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by Mygod** [comment]
Importing requires a good file manager, for example try MiXplorer.

**Comment by supernovae01** [comment]
> Importing requires a good file manager, for example try MiXplorer.

Still not able to add server profile. Tried both MiXplorer and ES File Explorer.

Manual function is more versatile imo. Is it going to be add back at some point?

**Comment by Mygod** [comment]
You need to use content URI while importing. If that doesn't work, export logcat and put it here.

If you prefer manual configuration, you can use the standard app on your TV.

**Comment by wenhaoy** [comment]
What's the format of the file?

**Comment by Mygod** [comment]
[Standard shadowsocks JSON format](https://shadowsocks.org/en/config/quick-guide.html). This object can appear anywhere in the file and batch import is also supported. Therefore, you can import things like `gui-config.json` from shadowsocks-windows.

However, note that for now you have to use a Android standard-compliant file manager that exports the file as a `content://` URI. Legacy `file://` URIs (which unfortunately a lot of file managers use) are not supported yet due to permission issues. Further improvements are welcome.

**Comment by zpfsqzyl** [comment]
You can use RE explorer from goole play store to import your profile.json , i succeed importing just now.

**Comment by psamim** [comment]
I have the same problem too. I ended up using the non-TV version.

**Comment by AlexMofer** [comment]
这个问题依然存在。无法选择文件载入，整个程序就废了。
致开发者：
虽然DocumentsProvider兼容性最佳（相对于手机来说），但是Android原生的DocumentsProvider实现，其实是针对手机设备的，对TV友好度极差，无法选择、无法点击的情况比比皆是。目前本人的就是无法定位到文件，更无法选择该文件。
建议：
1.将发送ACTION_OPEN_DOCUMENT的Intent修改为ACTION_OPEN_DOCUMENT_TREE，获取文档会导致进入最近列表而无法修改，从而毫无用处。而获取文件树则会进入磁盘根目录，此时用户选择文件所在目录，授权访问目录后，再在程序中列出json配置文件，让用户选择载入即可。
2.增加对“.json”文件的打开做支持，例如：
```
<activity
            android:name=".OpenActivity"
            android:autoRemoveFromRecents="true"
            android:documentLaunchMode="intoExisting"
            android:excludeFromRecents="true"
            android:exported="true">
            <intent-filter>
                <action android:name="android.intent.action.VIEW" />

                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />

                <data android:scheme="file" />
                <data android:scheme="content" />
                <data android:mimeType="application/json" />
            </intent-filter>
        </activity>
```
本人正在尝试写一个简单应用，将要配置的文件刷到最近列表中，看能不能正常使用。

**Comment by AlexMofer** [comment]
由Intent ACTION_GET_CONTENT或者ACTION_OPEN_DOCUMENT启动的选择文件的系统UI，无法被任何其他第三方APP所替代，支持DocumentsProvider的第三方文件管理器只会被列在系统UI的右侧边栏的底部，也无法被选用。任何非多媒体文件（图片、视频、音频）无法被添加到最近列表中。该问题无法从软件层面绕过。只能物理破局，买了个飞鼠，解决了问题。
期待开发者能尽早解决。
另外，创建配置文件也存在类似的问题。只不过该功能使用率极低。

**Comment by madeye** [comment]
Any PR is welcome.

**Comment by Mygod** [comment]
@AlexMofer Please report to AOSP. This is not an issue of our side.

**Comment by Mygod** [comment]
Fixed via e4557b952c852331cc79fa0c8e2ef7395fa6f7ae.
